### ISSUE TITLE
Incorrect return type in ast.NodeTransformer.generic_visit

### ISSUE THREAD

**Issue by Dentosal** [python/typeshed#2485]
`NodeTransformer.generic_visit` returns None or modified AST. (In [stdlib/3/ast.pyi](https://github.com/python/typeshed/blob/de1755b263ed0b80aba35f043550c91a508c3d08/stdlib/3/ast.pyi)). ([CPython source for ast.py](https://github.com/python/cpython/blob/3.7/Lib/ast.py#L311))

```python
class NodeTransformer(NodeVisitor):
    def generic_visit(self, node: AST) -> Optional[AST]: ...
```

It should probably look like this, but then tests do not pass, because the parent type (`NodeVisitor`) always returns `None`, and is marked as such.

**Comment by srittau** [comment]
We welcome pull requests. I think `NodeVisitor.generic_visit()` needs to be fixed to return `Any` like `visit()` as that seems to be the intent.

**Comment by Dentosal** [comment]
Thanks. If changing the return types to `Any` is acceptable instead of a more fine-grained solution, then I'm going to submit a PR.

**Comment by srittau** [comment]
Fixed with #2486 .
